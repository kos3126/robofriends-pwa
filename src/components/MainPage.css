/* #### Generated By: http://www.cufonfonts.com #### */

@font-face {
  font-display: swap;
  /* default text without the special font until the font is fully loaded */
  font-family: "SEGA LOGO FONT";
  font-style: normal;
  font-weight: normal;
  src: local("SEGA LOGO FONT"), url("../containers/SEGA.woff") format("woff");
}

h1 {
  font-family: "SEGA LOGO FONT";
  font-weight: 200;
  color: #0ccac4;
}
