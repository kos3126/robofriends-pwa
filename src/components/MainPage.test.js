import { shallow } from "enzyme";
import MainPage from "../components/MainPage";

describe("<MainPage />", () => {
  let wrapper,
    mockProps = {};

  beforeEach(() => {
    // it()やtest()毎に下記の状態に初期化される
    mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: "",
      isPending: false,
    };
    wrapper = shallow(<MainPage {...mockProps} />);
  });

  it("renders MainPage without crashing", () => {
    console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a loading text while pending", () => {
    Object.assign(mockProps, { isPending: true });
    wrapper = shallow(<MainPage {...mockProps} />);

    expect(wrapper.contains(<h1>Loading...</h1>)).toEqual(true);
  });

  it("filters robots correctly", () => {
    Object.assign(mockProps, {
      robots: [
        {
          id: 3,
          name: "John",
          email: "john@email.com",
        },
      ],
      searchField: "John",
    });
    wrapper = shallow(<MainPage {...mockProps} />);

    expect(wrapper.instance().filteredRobots()[0].name).toEqual("John");
    expect(wrapper.instance().filteredRobots()[0].id).toEqual(3);
    expect(wrapper.instance().filteredRobots()).toEqual([
      {
        id: 3,
        name: "John",
        email: "john@email.com",
      },
    ]);

    Object.assign(mockProps, {
      searchField: "a",
    });
    const filteredRobots = [];
    wrapper = shallow(<MainPage {...mockProps} />);

    expect(wrapper.instance().filteredRobots()).toEqual(filteredRobots);
  });
});
