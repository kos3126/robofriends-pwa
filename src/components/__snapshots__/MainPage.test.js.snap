// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
